### PR TITLE
Client registration summary

### PR DESCRIPTION
**Pull request by mmadariaga**
Added an endpoint into brand API that shows current registered client stats

#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [ ] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX)

### CODE REVIEW

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/DomainDoctrineRepository.php [hunk 1]
```diff
@@ -34,4 +47,53 @@ public function findOneByDomain($domain)
...
+                $company->getId()
+            );
+
+            if (!$companyDomain) {
+                continue;
+            }
+            $domains[] = $companyDomain;
+        }
+
+        return $domains;
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/DomainDoctrineRepository.php:82]: array_unique

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/DomainDoctrineRepository.php [hunk 2]
```diff
@@ -34,4 +47,53 @@ public function findOneByDomain($domain)
...
+        if (!$brand) {
+            return $domains;
+        }
+
+        $brandDomain = $brand->getDomain();
+        if ($brandDomain) {
+            $domains[] = $brandDomain;
+        }
+
+        $companies = $brand->getCompanies();
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/DomainDoctrineRepository.php:70]: filter by vpbx

File: web/rest/brand/src/Controller/My/ClientRegistrationSummaryAction.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        $userLocations = $this->usersLocationRepository->findByDomains(
+            $domainFqdns
+        );
+
+        $now = new \DateTime();
+        $active = 0;
+        $total = 0;
+
+        foreach ($userLocations as $userLocation) {
+            $total++;
```
Comment (mmadariaga) [inline web/rest/brand/src/Controller/My/ClientRegistrationSummaryAction.php:65]: Total must be calculated by a counter by entity matching criteria

**Comment by ironArt3mis** [comment]
Rebase completed.

File: library/Ivoz/Kam/Infrastructure/Persistence/Doctrine/UsersLocationDoctrineRepository.php [hunk 4]
```diff
@@ -56,4 +56,43 @@ public function findByUsernameAndDomain(string $username, string $domain): array
...
+     * @return UsersLocationInterface[]
+     */
+    public function findByDomains(array $domains): array
+    {
+        $qb = $this->createQueryBuilder('self');
+        $qb
+            ->select('self')
+            ->addCriteria(
+                CriteriaHelper::fromArray([
+                    [ 'domain', 'in', $domains ],
```
Comment (mmadariaga) [inline library/Ivoz/Kam/Infrastructure/Persistence/Doctrine/UsersLocationDoctrineRepository.php:71]: Skip expired

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/ResidentialDeviceDoctrineRepository.php [hunk 5]
```diff
@@ -32,6 +36,17 @@ public function findOneByNameAndDomain(string $name, DomainInterface $domain)
...
             "domain" => $domain
         ]);
+
         return $response;
     }
+
+    public function countRegistrableDevicesByBrand(int $brandId): int
+    {
+        $criteria = CriteriaHelper::fromArray([
+            ['brand', 'eq', $brandId],
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/ResidentialDeviceDoctrineRepository.php:46]: replace by comanyIds

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/RetailAccountDoctrineRepository.php [hunk 6]
```diff
@@ -32,6 +36,17 @@ public function findOneByNameAndDomain(string $name, DomainInterface $domain)
...
             "domain" => $domain
         ]);
+
         return $response;
     }
+
+    public function countRegistrableDevicesByBrand(int $brandId): int
+    {
+        $criteria = CriteriaHelper::fromArray([
+            ['brand', 'eq', $brandId],
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/RetailAccountDoctrineRepository.php:46]: replace by companyIds

File: web/rest/brand/config/api/resources.yml [hunk 7]
```diff
@@ -48,3 +47,18 @@ Model\ActiveCalls:
           required: false
           description: 'Company Id'
   collectionOperations: []
+
+Model\RegistrationSummary:
+  itemOperations: []
+  collectionOperations:
+    get_my_registration_summary:
+      access_control: '"ROLE_BRAND_ADMIN" in roles && user.hasAccessPrivileges(_api_resource_class, request.getMethod())'
```
Comment (mmadariaga) [inline web/rest/brand/config/api/resources.yml:55]: Add it into ACLs

File: web/rest/brand/src/Controller/My/RegistrationSummaryAction.php [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+            $domainFqdns
+        );
+
+        $now = new \DateTime();
+        $active = 0;
+        $total = $this->getTotalDevices(
+            $brand->getId()
+        );
+
+        foreach ($userLocations as $userLocation) {
```
Comment (mmadariaga) [inline web/rest/brand/src/Controller/My/RegistrationSummaryAction.php:86]: this goes on repository

File: web/rest/brand/src/Model/RegistrationSummary.php [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+    /**
+     * @var integer
+     * @AttributeDefinition(type="int")
+     */
+    protected $total;
+
+    /**
+     * @var float
+     * @AttributeDefinition(type="float")
```
Comment (mmadariaga) [inline web/rest/brand/src/Model/RegistrationSummary.php:26]: int

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/DomainDoctrineRepository.php [hunk 10]
```diff
@@ -34,4 +48,60 @@ public function findOneByDomain($domain)
...
+    /**
+     * Includes company domains
+     * @param int $brandId
+     * @return DomainInterface[]
+     */
+    public function findByBrandId(int $brandId): array
+    {
+        $domains = [];
+
+        /** @var BrandInterface $brand */
```
Comment (Kaian) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/DomainDoctrineRepository.php:61]: BrandInterface | null ?

File: library/DataFixtures/ORM/ProviderAdministrator.php [hunk 11]
```diff
@@ -138,6 +138,39 @@ public function load(ObjectManager $manager)
...
+            $this->setPass("changeme");
+            $this->setEmail("test@irontec.com");
+            $this->setActive(true);
+            $this->setName("Admin Name");
+            $this->setLastname("Admin Lastname");
+            $this->setBrand($fixture->getReference('_reference_ProviderBrand1'));
+            $this->setCompany($fixture->getReference('_reference_ProviderCompany3'));
+            $this->setTimezone($fixture->getReference('_reference_ProviderTimezone145'));
+        })->call($item9);
+        $this->addReference('_reference_ProviderAdministrator9', $item7);
```
Comment (Kaian) [inline library/DataFixtures/ORM/ProviderAdministrator.php:170]: $item9 here

File: library/DataFixtures/ORM/ProviderAdministrator.php [hunk 12]
```diff
@@ -138,6 +138,39 @@ public function load(ObjectManager $manager)
...
+            $this->setPass("changeme");
+            $this->setEmail("test@irontec.com");
+            $this->setActive(true);
+            $this->setName("Admin Name");
+            $this->setLastname("Admin Lastname");
+            $this->setBrand($fixture->getReference('_reference_ProviderBrand1'));
+            $this->setCompany($fixture->getReference('_reference_ProviderCompany4'));
+            $this->setTimezone($fixture->getReference('_reference_ProviderTimezone145'));
+        })->call($item8);
+        $this->addReference('_reference_ProviderAdministrator8', $item7);
```
Comment (Kaian) [inline library/DataFixtures/ORM/ProviderAdministrator.php:153]: $item8 here

File: library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/DomainDoctrineRepository.php [hunk 13]
```diff
@@ -34,4 +48,60 @@ public function findOneByDomain($domain)
...
+        return $domains;
+    }
+
+    /**
+     * @param int $companyId
+     * @return DomainInterface|null
+     */
+    public function findByCompanyId(int $companyId)
+    {
+        /** @var CompanyInterface $company */
```
Comment (Kaian) [inline library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/DomainDoctrineRepository.php:99]: CompanyInterface | null

File: web/rest/brand/src/Controller/My/RegistrationSummaryAction.php [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+        $this->companyRepository = $companyRepository;
+        $this->terminalRepository = $terminalRepository;
+        $this->friendRepository = $friendRepository;
+        $this->residentialDeviceRepository = $residentialDeviceRepository;
+        $this->retailAccountRepository = $retailAccountRepository;
+    }
+
+    public function __invoke()
+    {
+        /** @var TokenInterface $token */
```
Comment (Kaian) [inline web/rest/brand/src/Controller/My/RegistrationSummaryAction.php:52]: Phpstan Lv4 will complain

**Pull request merged**